packages/jest-runtime/src/index.ts

```diff
@@ -688,14 +688,19 @@ export default class Runtime {
     };
     moduleRegistry.set(modulePath, localModule);
 
-    this._loadModule(
-      localModule,
-      from,
-      moduleName,
-      modulePath,
-      options,
-      moduleRegistry,
-    );
+    try {
+      this._loadModule(
+        localModule,
+        from,
+        moduleName,
+        modulePath,
+        options,
+        moduleRegistry,
+      );
+    } catch (error: unknown) {
+      moduleRegistry.delete(modulePath);
```

Comment:
shouldn't we do the same for mockRegistry?

Reply:
good point! we should 😀

Reply:
we already do actually

https://github.com/facebook/jest/blob/57e32e99a7994dfdcca8f777a2961294919551b9/packages/jest-runtime/src/index.ts#L812